src/devices/Qmc5883l/VectorExtension.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Numerics;
+using UnitsNet;
+
+namespace Iot.Device.Qmc5883l
+{
+    /// <summary>
+    /// Extension class for getting a heading from X and Y Vectors.
+    /// </summary>
+    public static class VectorExtension
+    {
+        /// <summary>
+        /// Calculate heading.
+        /// </summary>
+        /// <param name="vector">Direction Vector</param>
+        /// <returns>Heading (Angle)</returns>
+        public static Angle GetHeading(this Vector3 vector)
+        {
+            double deg = Math.Atan2(vector.Y, vector.X) * 180 / Math.PI;
+
+            deg = deg < 0 ? deg + 360 : deg;
+            return Angle.FromDegrees(deg);
+        }
+    }
```

Comment:
I'd recommend to only add this in the sample. Especially this has high likelihood of someone doing exactly same in their code which would make it very annoying to use if that happens.